### PR TITLE
Fix the identity operator id in AA

### PR DESCRIPTION
**Pull request by stiepan**
## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
Tl;dr: for the end user, the AA works corectly, but under the hood there is a bug in handling sub-policies of different length that is hidden by a happy coincidence.

Sub-policies in AA can have different length. When preparing the matrix that describes mapping from sub-policy and stage to augmentation to run, the shorter sub-polciies are padded with the identity operator. However, some time ago, https://github.com/NVIDIA/DALI/pull/4751 intrduced a bug hat uses index out of range instead of the index of the identity operator.

Fortunately, it has no implications for correctness so far because: the `select` helper runs the last augmentation in the list if the index is out of range and the last augmentation in the list is always the identity.

This PR fixes the out-of-range index and adds unit test to the function that prepares the mapping matrix.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by stiepan** [comment]
!build

File: dali/python/nvidia/dali/auto_aug/auto_augment.py [hunk 1]
```diff
@@ -477,7 +478,7 @@ def _sub_policy_to_augmentation_map(policy: Policy) -> Tuple[_DataNode, List[Lis
                     stage_augments.add(aug)
                     stage_augments_list.append(aug)
         augmentations.append(stage_augments_list + [a.identity])
-    identity_id = [len(stage_augments) for stage_augments in augmentations]
+    identity_id = [len(stage_augments) - 1 for stage_augments in augmentations]
```
Comment (stiepan) [inline dali/python/nvidia/dali/auto_aug/auto_augment.py:481]: This is the fix

**Comment by stiepan** [comment]
!build

**Review by szalpal — APPROVED** [review]

**Review by jantonguirao — APPROVED** [review]
